src/searchModal.js

```diff
@@ -98,16 +119,106 @@ export default function searchModalFactory(config) {
             .focus();
     }
 
+    /**
+     * Inits class filter selector
+     */
+    function initClassFilter() {
+        return new Promise(function (resolve) {
+            const rootClassUri = instance.config.rootClassUri;
+            const initialClassUri =
+                instance.config.criterias && instance.config.criterias.class
+                    ? instance.config.criterias.class
+                    : instance.config.rootClassUri;
+            resourceSelector = resourceSelectorFactory($('.class-tree', instance.getElement()), {
+                //set up the inner resource selector
+                selectionMode: 'single',
+                selectClass: true,
+                classUri: rootClassUri,
+                showContext: false,
+                showSelection: false
+            });
+
+            // when a class query is triggered, update selector options with received resources
+            resourceSelector.on('query', function (params) {
+                if (instance.config.testMocks.classTree) {
+                    resourceSelector.update(instance.config.testMocks.classTree, params);
+                    return;
+                }
+                params.classOnly = true;
```

Comment:
Arguments should not be muted, this may introduce unexpected behavior.
If you need to adapt the input, please do it in a copy